source/string.c

```diff
@@ -15,12 +15,16 @@
 #include <aws/common/string.h>
 
 struct aws_string *aws_string_new_from_c_str(struct aws_allocator *allocator, const char *c_str) {
+    if (c_str == NULL || allocator == NULL) {
+        return NULL;
+    }
     return aws_string_new_from_array(allocator, (const uint8_t *)c_str, strlen(c_str));
 }
 
 struct aws_string *aws_string_new_from_array(struct aws_allocator *allocator, const uint8_t *bytes, size_t len) {
+    AWS_PRECONDITION(AWS_MEM_IS_READABLE(bytes, len));
     size_t malloc_size;
-    if (aws_add_size_checked(sizeof(struct aws_string) + 1, len, &malloc_size)) {
+    if (aws_add_size_checked(sizeof(struct aws_string) + 1, len, &malloc_size) || !allocator || !bytes) {
```

Comment:
Should I return `NULL` in case `allocator` or `bytes` are equal to NULL, or just check those in `AWS_FATAL_ASSERT`?

Reply:
I think option 2, a FATAL_PRECONDITION is probably the right call here